api/app/api.py

```diff
@@ -19,7 +41,105 @@
     allow_headers=["*"]
 )
 
+class ForecastRequest(BaseModel):
+    site: PVSite
+    timestamp: Optional[str] = None
+    nwp_source: Optional[str] = "icon"
+    access_token: Optional[str] = None
+    enphase_system_id: Optional[str] = None
+
+class AuthUrlRequest(BaseModel):
+    full_auth_url: str
+
 @app.post("/forecast/")
-async def forecast(site:PVSite):
-    df =run_forecast(site)
-    return df.to_dict()
\ No newline at end of file
+async def forecast(request: ForecastRequest):
+    site = request.site
+    ts = request.timestamp if request.timestamp else pd.Timestamp.now(tz='UTC').isoformat()
+    nwp_source = request.nwp_source
+    access_token = request.access_token
+    enphase_system_id = request.enphase_system_id
+
+    try:
+        logging.info(f"Received forecast request: {request}")
+
+        timestamp = pd.Timestamp(ts).tz_localize(None)
+        formatted_timestamp = timestamp.strftime('%Y-%m-%d %H:%M:%S')
+
+        logging.info(f"Processed forecast request: site={site} timestamp={formatted_timestamp} nwp_source={nwp_source} enphase_system_id={enphase_system_id}")
+
+        solis_data = None
+        givenergy_data = None
+        
+        if site.inverter_type:
```

Comment:
I am not quite sure about this part here, we are getting the inverter data here but they can also be got in make_pv_data later?

Reply:
> I am not quite sure about this part here, we are getting the inverter data here but they can also be got in make_pv_data later?

this is the prefetch code
so to break this down, tmake_pv_data can fetch data when called from cli and no params(see optional params of that func) are provided - this is in the CLI context
but when the data is provided as params(prefetched data), it can be used in the API context beofr even reaching the CLI part

Reply:
hmm, i think i agree with @Sukh-P, would it not be simplier if it all goes in make_pv_data later

Reply:
> hmm, i think i agree with @Sukh-P, would it not be simplier if it all goes in `make_pv_data` later

it woudlve made sense if some of the functions didnt ask for an auth url in the CLI- which we cant afford to have in the website

Reply:
see other comments, I think we are talking about all the same thing